# Import <br>

Các thư viện cần thiết cho bài làm này

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns; sns.set()
pd.set_option('display.max_columns', None)  

Điều chỉnh kích thước và độ phóng chung cho các biểu đồ: figsize = (10,5) ; dpi = 80

In [ ]:
plt.rcParams['figure.figsize'] = [10, 5]
plt.rcParams['figure.dpi'] = 80

# Khám phá dữ liệu

### Đọc dữ liệu

In [10]:
EPL = pd.read_excel('Candidate Data/EPL.xlsx')
EPL.head()

Matchweek  Day       Date     Season   Time         home_team  home_score  \
0          1  Fri 2022-08-05  2022-2023  20:00    Crystal Palace           0   
1          1  Sat 2022-08-06  2022-2023  12:30            Fulham           2   
2          1  Sat 2022-08-06  2022-2023  15:00         Tottenham           4   
3          1  Sat 2022-08-06  2022-2023  15:00  Newcastle United           2   
4          1  Sat 2022-08-06  2022-2023  15:00      Leeds United           2   

           away_team  away_score                      Venue         Referee  \
0            Arsenal           2              Selhurst Park  Anthony Taylor   
1          Liverpool           2             Craven Cottage     Andy Madley   
2        Southampton           1  Tottenham Hotspur Stadium  Andre Marriner   
3  Nottingham Forest           0             St James' Park    Simon Hooper   
4             Wolves           1                Elland Road    Robert Jones   

   diff_score  
0          -2  
1           0  
2           3  
3           2  
4           1

In [11]:
ITA = pd.read_excel('Candidate Data/ITA.xlsx')
ITA.head()

Matchweek  Day       Date     Season   Time   home_team  home_score  \
0        1.0  Sat 2022-08-13  2022-2023  18:30   Sampdoria           0   
1        1.0  Sat 2022-08-13  2022-2023  18:30       Milan           4   
2        1.0  Sat 2022-08-13  2022-2023  20:45       Monza           1   
3        1.0  Sat 2022-08-13  2022-2023  20:45       Lecce           1   
4        1.0  Sun 2022-08-14  2022-2023  18:30  Fiorentina           3   

   away_team  away_score                           Venue              Referee  \
0   Atalanta           2  Stadio Comunale Luigi Ferraris     Federico Dionisi   
1    Udinese           2          Stadio Giuseppe Meazza      Livio Marinelli   
2     Torino           2                 U-Power Stadium     Maurizio Mariani   
3      Inter           2    Stadio Comunale Via Del Mare  Alessandro Prontera   
4  Cremonese           2          Stadio Artemio Franchi          Juan Sacchi   

   diff_score  
0          -2  
1           2  
2          -1  
3          -1  
4           1

### Dữ liệu có bao nhiêu dòng và cột ?

In [12]:
n_rows, n_col = EPL.shape
n_rows, n_col

(663, 12)

In [13]:
n_rows, n_col = ITA.shape
n_rows, n_col

(670, 12)

Như vậy ta có được dữ liệu EPL hiện tại đưa vào là 663 dòng và 12 cột.
Như vậy ta có được dữ liệu EPL hiện tại đưa vào là 670 dòng và 12 cột.

### Các dòng dữ liệu

#### Mỗi dòng có ý nghĩa gì? Có vấn đề các dòng có ý nghĩa khác nhau không?

In [14]:
EPL.sample(3)

Matchweek  Day       Date     Season   Time  home_team  home_score  \
432          6  Sat 2023-09-23  2023-2024  17:30  Brentford           1   
242         26  Sat 2023-03-04  2022-2023  15:00    Arsenal           3   
202         22  Sat 2023-02-04  2022-2023  15:00     Wolves           3   

       away_team  away_score                    Venue         Referee  \
432      Everton           3  Gtech Community Stadium  Michael Oliver   
242  Bournemouth           2         Emirates Stadium  Chris Kavanagh   
202    Liverpool           0         Molineux Stadium    Paul Tierney   

     diff_score  
432          -2  
242           1  
202           3

### Dữ liệu các dòng có bị lặp hay không?

In [15]:

EPL[EPL.duplicated(keep=False)]

Empty DataFrame
Columns: [Matchweek, Day, Date, Season, Time, home_team, home_score, away_team, away_score, Venue, Referee, diff_score]
Index: []

In [16]:

ITA[ITA.duplicated(keep=False)]

Empty DataFrame
Columns: [Matchweek, Day, Date, Season, Time, home_team, home_score, away_team, away_score, Venue, Referee, diff_score]
Index: []

### Mỗi cột hiện đang có kiểu dữ liệu gì? Có cột nào có kiểu dữ liệu chưa phù hợp để có thể xử lý tiếp không?

In [17]:
EPL.dtypes

Matchweek              int64
Day                   object
Date          datetime64[ns]
Season                object
Time                  object
home_team             object
home_score             int64
away_team             object
away_score             int64
Venue                 object
Referee               object
diff_score             int64
dtype: object

In [18]:
ITA.dtypes

Matchweek            float64
Day                   object
Date          datetime64[ns]
Season                object
Time                  object
home_team             object
home_score             int64
away_team             object
away_score             int64
Venue                 object
Referee               object
diff_score             int64
dtype: object

Cột Matchweek của ITA sẽ được chuyển về kiểu dữ liệu int
=> Không thể chuyển về dạng int vì dữ liệu có chứa nan value

In [30]:
EPL.describe()

Matchweek  home_score  away_score  diff_score
count  663.000000  663.000000  663.000000  663.000000
mean    17.435897    1.687783    1.330317    0.357466
std     10.169190    1.379813    1.235588    1.965057
min      1.000000    0.000000    0.000000   -8.000000
25%      9.000000    1.000000    0.000000   -1.000000
50%     17.000000    1.000000    1.000000    0.000000
75%     25.000000    2.000000    2.000000    2.000000
max     38.000000    9.000000    8.000000    9.000000